### PR TITLE
codec/table: refactor encode_row_key

### PR DESCRIPTION
**Pull request by AndreMouche**
## What have you changed? (mandatory)

Now the function in master is defined as follows:

```
/// `encode_row_key` encodes the table id and record handle into a byte array.
pub fn encode_row_key(table_id: i64, encoded_handle: &[u8]) -> Vec<u8>
```

And this PR defined it like following to make it more easy to use

```
/// `encode_row_key` encodes the table id and record handle into a byte array.
pub fn encode_row_key(table_id: i64, handle: i64) -> Vec<u8>
```

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Refer to a related PR or issue link (optional)

https://github.com/pingcap/tikv/issues/3157

@BusyJay @breeswish PTAL

### CODE REVIEW

File: benches/bin/utils.rs [hunk 1]
```diff
@@ -29,9 +29,7 @@ pub fn next_ts() -> u64 {
 pub fn generate_row_keys(table_id: i64, start_id: i64, count: usize) -> Vec<Vec<u8>> {
     let mut result = Vec::with_capacity(count);
     for i in (start_id)..(start_id + count as i64) {
-        let mut handle = Vec::with_capacity(8);
-        handle.encode_i64(i as i64).unwrap();
-        let key = encode_row_key(table_id, &handle);
+        let key = encode_row_key(table_id, i as i64);
```
Comment (BusyJay) [inline benches/bin/utils.rs]: Why cast?

**Comment by AndreMouche** [comment]
ping @breeswish

**Comment by breeswish** [comment]
LGTM

**Pull request merged**